### DISCUSSION TITLE
ForwardMessagesAsync Gives me an invalid id of the sent message

### DISCUSSION THREAD

**Discussion by Shpakoff** [egramtel/tdsharp#160] [Q&A] [answered]
Client.ForwardMessagesAsync, somehow works strangely, why do I get the id of a message that, just by one more than the id of the previous message from the chat. And when I try to find this message in this chat through the received id, I get an error.
Example my code for get id:
var messages = await Client.ForwardMessagesAsync(
                        chatId: chatId,
                        fromChatId: fromChatId,
                        messageIds: new[] { postId },
                        sendCopy: true);
                    return messages.Messages_[0].Id;

=[20:24:44]m_id[151400742912] message      
=forward post -> [20:24:44] m_id[151400742913]  forward message

messages.Messages_[0].Id =  151400742913

when i get chathistory
=[20:24:44]m_id[151400742912] message    
=[20:24:44] m_id[151401791488]  forward message

Or should I calculate this id myself so 151400742913 - 1 + 1048576

**Comment by ForNeVeR** [comment, marked as answer]
I think that it would be better to [ask this from the upstream](https://github.com/tdlib/td/issues). I don't believe this is an issue of the API translation layer.

> Or should I calculate this id myself

I don't believe you should calculate any message id manually.

**Reply by Shpakoff** [reply]
Okay, I wrote to them, as soon as I find an answer, I'll write back.

**Reply by ForNeVeR** [reply]
https://github.com/tdlib/td/issues/2329 seems to contain the answer.
